myapp/utility.py

```diff
@@ -1,116 +1,880 @@
+                     'class_name': lab.class_name,
+                     'prof_name': lab.prof_name,
+                     'status': lab.status})
+    return labs
+
+
+# convert a query for experiments into a list of JSON objects
+def serialize_experiment_list(experiment_list_query):
+    experiments = []
+    for experiment in experiment_list_query:
+        experiments.append({'experiment_id': experiment.id,
+                            'experiment_name': experiment.name,
+                            'description': experiment.description,
+                            'order': experiment.order,
+                            'value_type': experiment.value_type,
+                            'value_range': experiment.value_range,
+                            'value_candidates': experiment.value_candidates})
+    return experiments
+
+
+# return a list of class names
+def serialize_class_name_list():
+    return [c.name for c in query_all_class()]
+
+
+# return a list of professors' names
+def serialize_prof_name_list():
+    return [u.name for u in query_all_admin()] + [u.name for u in query_all_superadmin()]
+
+
+# convert a query for admins into a list of JSON objects
+def serialize_admin_list(admins_query):
+    current_admins = []
+    for admin in admins_query:
+        classes = [c.id for c in admin.classes]
+        labs = [lab.id for lab in admin.labs]
+        current_admins.append({'id': admin.id,
+                               'username': admin.username,
+                               'password': admin.password,
+                               'name': admin.name, 'classes': classes,
+                               'labs': labs})
+    return current_admins
+
+
+# convert a query for classes into a list of JSON objects
+def serialize_class_list(classes_query):
+    current_classes = []
+    for c in classes_query:
+        professors = [p.username for p in c.users if (p.role_name == 'Admin' or p.role_name == 'SuperAdmin')]
+        students = [s.username for s in c.users if s.role_name == 'Student']
+        labs = [lab.id for lab in c.labs]
+        current_classes.append({'id': c.id,
+                                'name': c.name,
+                                'time': c.time,
+                                'professors': professors,
+                                'students': students,
+                                'labs': labs})
+    return current_classes
+
+
+# check the data sent by client and
+# deserialize the information of a lab from jsonData sent by client
+# and serialize it into JSON format dictionary that is consistent with
+# python side naming convention
+def deserialize_lab(lab_json):
+    err_msg = check_existence(lab_json, 'labName', 'className', 'classTime',
+                                        'professorName', 'labDescription',
+                                        'experiments')
+    if err_msg != '':
+        return (dict(), err_msg)
+
+    lab_info = {'name': lab_json['labName'],
```

Comment:
I will open an issue for it and probably rewrite it. The current implementation is with thought that I should try to make the serialize lab and deserialize_lab be the opposite of each other. Besides, there is a key-pair which is the value of old_lab_id which is not an attribute of the Lab class.